### ISSUE TITLE
Streamify creating invalid JSON

### ISSUE THREAD

**Issue by GlynnPhillips** [philbooth/bfj#25]
Afternoon @philbooth, I believe I have found an issue whereby invalid JSON will be produced. I think might be related to the usage of `streamify`.

Passing the following object to `bfj.write`

```
{
    "count":1,
    "email": undefined,      
    "wait":600
}
```

Will result in

```
{
    "count":1,
    "email":
    "wait":600
}
```

I am not sure how `streamify` should be handling `JSON` so am unsure if this is an issue that should be raised within their issues or if we can swap it out for a different module, what do you think?

**Comment by philbooth** [comment]
Thanks for the bug report @GlynnPhillips. It's a bona fide problem in bfj, `streamify` here isn't the module from npm it is a source module in this repo. Debugging now.

**Comment by GlynnPhillips** [comment]
Opps that will teach me for not reading the code fully and jumping to conclusions. Happy to also take a look tomorrow if you don't have time.

**Comment by philbooth** [comment]
Fixed and published to npm as `1.2.1`. Aren't I good! :grin:

The problem was a complete lack of handling for `undefined` values. I'm super glad you spotted it because my handling of functions and symbols was wrong too. New behaviour is as per `JSON.stringify`, so they're ignored in objects and coerced to `null` in arrays.

The fix, in case you're interested: https://github.com/philbooth/bfj/commit/4b5d9708ddf91d68b6c9c4b19dfbba70b93f83ca

**Comment by GlynnPhillips** [comment]
You are the best, thanks Phil
